pkg/apis/serving/k8s_validation.go

```diff
@@ -142,12 +161,20 @@ func validateVolume(ctx context.Context, volume corev1.Volume) *apis.FieldError
 		specified = append(specified, "emptyDir")
 		errs = errs.Also(validateEmptyDirFields(vs.EmptyDir).ViaField("emptyDir"))
 	}
+
+	if vs.PersistentVolumeClaim != nil {
+		specified = append(specified, "persistentVolumeClaim")
```

Comment:
I'm wondering if we should call `shouldAllowPersistentVolumeClaims->validatePersistentVolumeClaims` here (rather than in ValidateVolumes) to be consistent with the other volume types above

Reply:
Yeah makes sense will switch to that.

Reply:
did you miss this one?

Reply:
Ok let me check.

Reply:
@julz this part is about validating the volume itself wrt to K8s reqs. validatePersistentVolumeClaims is related to whether we should allow the specific type of volume wrt to the Serving feature. That is why I used the name  `shouldAllowPersistentVolumeClaims` similarly to what we have for emptydir.

Reply:
moved all volume related checks to the volumeValidate func